tutorials/tutorial-grafana.md

```diff
@@ -0,0 +1,196 @@
+
+Instead of using the Grafana query builder, we will edit our query directly. In the 
+view, click on the 'Edit SQL' button at the bottom.
+
+<img class="main-content__illustration" src="https://s3.amazonaws.com/docs.timescale.com/tutorial-grafana/edit_sql_in_grafana.png" alt="Edit custom SQL queries in Grafana"/>
+
+Before we can begin authoring our query, we also want to set the Query database to the New 
+York City taxi cab datasource we connected to earlier:
+
+<img class="main-content__illustration" src="https://s3.amazonaws.com/docs.timescale.com/tutorial-grafana/set_datasource.png" alt="Switching data sources in Grafana"/>
+
+### Visualize metrics stored in TimescaleDB
+
+Let’s start by creating a visualization that answers the question **How many rides took place on each day?** 
+from the [Hello, Timescale!][hello-timescale] tutorial.
+
+From the tutorial, you can see the standard SQL syntax for our query:
+
+```sql
+SELECT date_trunc('day', pickup_datetime) as day, 
+COUNT(*) 
+FROM rides 
+GROUP BY day 
+ORDER BY day;
+```
+
+We will need to alter this query to support Grafana’s unique query syntax.
+
+#### Modifying the SELECT statement
+
+First, we will modify the `date_trunc` function to use the TimescaleDB `time_bucket` 
+function. You can consult the TimescaleDB [API Reference on time_bucket][time-bucket-reference] 
+for more information on how to use it properly.
+
+Let’s examine the `SELECT` portion of this query. First, we will bucket our results into 
+one day groupings using the `time_bucket` function. If you set the 'Format' of a Grafana 
+panel to be 'Time series', for use in Graph panel for example, then the query must return 
+a column named `time` that returns either a SQL `datetime` or any numeric datatype 
+representing a Unix epoch.
+
+So, part 1 of this new query is modified so that the output of the `time_bucket` grouping 
+is labeled `time` as Grafana requires, while part 2 is unchanged:
+
+```sql
+SELECT 
+  --1--
+  time_bucket('1 day', pickup_datetime) AS "time",
+  --2--
+  COUNT(*)
+FROM rides 
+```
+
+#### The Grafana __timeFilter command
+
+Grafana time-series panels include a tool that enables the end-user to filter on a given 
+time range. A “time filter,” if you will. Not surprisingly, Grafana has a way to link the 
+user interface construct in a Grafana panel with the query itself. In this case, 
+the `$__timefilter()` function.
+
+In the modified query below, we will enable the Grafana filtering user interface element 
+to set the range for the `pickup_datetime` column in our original New York City taxicab 
+database.
+
+```sql
+SELECT 
+  --1--
+  time_bucket('1 day', pickup_datetime) AS "time",
+  --2--
+  count(*)
+FROM rides
+where $__timeFilter(pickup_datetime)
+```
+
+#### Referencing elements in our query
+
+Finally, Grafana enables us to reference elements of the `SELECT` statement 
```

Comment:
this is not specific to grafana but part of the postgresql SQL dialect, the query you are writing is normal postgres SQL and only the parts starting with `$__` use grafana specific functions